lib/rpmsg/rpmsg_virtio.c

```diff
@@ -478,10 +478,11 @@ static int rpmsg_virtio_ns_callback(struct rpmsg_endpoint *ept, void *data,
 	_ept = rpmsg_get_endpoint(rdev, name, RPMSG_ADDR_ANY, dest);
 
 	if (ns_msg->flags & RPMSG_NS_DESTROY) {
-		if (_ept) {
-			metal_mutex_release(&rdev->lock);
-			rpmsg_destroy_ept(_ept);
-		}
+		if (_ept)
+			_ept->dest_addr = RPMSG_ADDR_ANY;
```

Comment:
why setting the dest_addr to ANY?

Reply:
if application don't destroy the endpoint, this allows to reconnect the end point on a new  ns announcement ( for instance crash usecase)